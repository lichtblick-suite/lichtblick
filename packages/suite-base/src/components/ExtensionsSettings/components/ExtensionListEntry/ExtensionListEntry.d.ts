/// <reference types="react" />
import { Immutable } from "@lichtblick/suite";
import { ExtensionMarketplaceDetail } from "@lichtblick/suite-base/context/ExtensionMarketplaceContext";
type Props = {
    entry: Immutable<ExtensionMarketplaceDetail>;
    onClick: () => void;
    searchText: string;
};
export default function ExtensionListEntry({ entry: { id, description, name, publisher, version }, searchText, onClick, }: Props): React.JSX.Element;
export {};
