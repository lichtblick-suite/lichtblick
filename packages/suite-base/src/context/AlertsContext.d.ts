/// <reference types="react" />
import { StoreApi } from "zustand";
import { Immutable } from "@lichtblick/suite";
import { PlayerAlert } from "@lichtblick/suite-base/players/types";
/**
 * セッションアラートの型定義
 *
 * プレイヤーアラートと同じ構造
 */
export type SessionAlert = PlayerAlert;
/**
 * タグ付きアラートの型定義
 *
 * アラートにタグを追加して識別可能にする
 */
type TaggedAlert = SessionAlert & {
    tag: string;
};
/**
 * アラートContextストアの型定義
 *
 * アラート一覧とアクションを含む
 */
export type AlertsContextStore = Immutable<{
    alerts: TaggedAlert[];
    actions: {
        clearAlert: (tag: string) => void;
        setAlert: (tag: string, alert: Immutable<SessionAlert>) => void;
    };
}>;
/**
 * ## AlertsContext
 *
 * **アラート管理のContext**
 *
 * ### 概要
 * - アプリケーション全体のアラート表示を管理
 * - Zustandストアを使用した状態管理
 * - タグベースのアラート識別
 *
 * ### 使用例
 * ```typescript
 * const actions = useAlertsActions();
 *
 * // アラート設定
 * actions.setAlert("error", {
 *   message: "エラーが発生しました",
 *   severity: "error"
 * });
 *
 * // アラート削除
 * actions.clearAlert("error");
 * ```
 */
export declare const AlertsContext: import("react").Context<StoreApi<{
    readonly alerts: readonly {
        readonly severity: import("../util/sendNotification").NotificationSeverity;
        readonly message: string;
        readonly error?: {
            readonly name: string;
            readonly message: string;
            readonly stack?: string | undefined;
            readonly cause?: unknown;
        } | undefined;
        readonly tip?: string | undefined;
        readonly tag: string;
    }[];
    readonly actions: {
        readonly clearAlert: (tag: string) => void;
        readonly setAlert: (tag: string, alert: Immutable<SessionAlert>) => void;
    };
}> | undefined>;
/**
 * アラートストアから値を取得するカスタムフック
 *
 * @param selector - ストアから値を選択する関数
 * @returns T - 選択された値
 */
export declare function useAlertsStore<T>(selector: (store: AlertsContextStore) => T): T;
/**
 * アラートアクションを取得するカスタムフック
 *
 * @returns AlertsContextStore["actions"] - アラート操作アクション
 */
export declare function useAlertsActions(): AlertsContextStore["actions"];
export {};
