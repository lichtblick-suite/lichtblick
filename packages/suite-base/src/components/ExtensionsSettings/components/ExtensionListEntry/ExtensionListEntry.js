import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
// SPDX-FileCopyrightText: Copyright (C) 2023-2025 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)<lichtblick@bmwgroup.com>
// SPDX-License-Identifier: MPL-2.0
import { ListItem, ListItemButton, ListItemText, Typography } from "@mui/material";
import Stack from "@lichtblick/suite-base/components/Stack";
import TextHighlight from "@lichtblick/suite-base/components/TextHighlight";
import { useStyles } from "./ExtensionListEntry.style";
export default function ExtensionListEntry({ entry: { id, description, name, publisher, version }, searchText, onClick, }) {
    const { classes } = useStyles();
    return (_jsx(ListItem, { disablePadding: true, "data-testid": "extension-list-entry", children: _jsx(ListItemButton, { className: classes.listItemButton, onClick: onClick, children: _jsx(ListItemText, { disableTypography: true, primary: _jsxs(Stack, { direction: "row", alignItems: "baseline", gap: 0.5, children: [_jsx(Typography, { variant: "subtitle2", fontWeight: 600, children: _jsx(TextHighlight, { targetStr: name, searchText: searchText }) }), _jsx(Typography, { variant: "caption", color: "text.secondary", children: version })] }), secondary: _jsxs(Stack, { gap: 0.5, children: [_jsx(Typography, { variant: "body2", color: "text.secondary", children: description }), _jsx(Typography, { color: "text.primary", variant: "body2", children: publisher })] }) }) }) }, id));
}
