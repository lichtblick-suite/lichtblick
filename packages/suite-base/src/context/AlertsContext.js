// SPDX-FileCopyrightText: Copyright (C) 2023-2025 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)<lichtblick@bmwgroup.com>
// SPDX-License-Identifier: MPL-2.0
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/
import { createContext } from "react";
import { useStore } from "zustand";
import { useGuaranteedContext } from "@lichtblick/hooks";
/**
 * ## AlertsContext
 *
 * **アラート管理のContext**
 *
 * ### 概要
 * - アプリケーション全体のアラート表示を管理
 * - Zustandストアを使用した状態管理
 * - タグベースのアラート識別
 *
 * ### 使用例
 * ```typescript
 * const actions = useAlertsActions();
 *
 * // アラート設定
 * actions.setAlert("error", {
 *   message: "エラーが発生しました",
 *   severity: "error"
 * });
 *
 * // アラート削除
 * actions.clearAlert("error");
 * ```
 */
export const AlertsContext = createContext(undefined);
AlertsContext.displayName = "AlertsContext";
/**
 * アラートストアから値を取得するカスタムフック
 *
 * @param selector - ストアから値を選択する関数
 * @returns T - 選択された値
 */
export function useAlertsStore(selector) {
    const context = useGuaranteedContext(AlertsContext);
    return useStore(context, selector);
}
const selectActions = (store) => store.actions;
/**
 * アラートアクションを取得するカスタムフック
 *
 * @returns AlertsContextStore["actions"] - アラート操作アクション
 */
export function useAlertsActions() {
    return useAlertsStore(selectActions);
}
