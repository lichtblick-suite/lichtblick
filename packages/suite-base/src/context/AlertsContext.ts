// SPDX-FileCopyrightText: Copyright (C) 2023-2025 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)<lichtblick@bmwgroup.com>
// SPDX-License-Identifier: MPL-2.0

// This Source Code Form is subject to the terms of the Mozilla Public
// License, v2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/

import { createContext } from "react";
import { StoreApi, useStore } from "zustand";

import { useGuaranteedContext } from "@lichtblick/hooks";
import { Immutable } from "@lichtblick/suite";
import { PlayerAlert } from "@lichtblick/suite-base/players/types";

/**
 * セッションアラートの型定義
 *
 * プレイヤーアラートと同じ構造
 */
export type SessionAlert = PlayerAlert;

/**
 * タグ付きアラートの型定義
 *
 * アラートにタグを追加して識別可能にする
 */
type TaggedAlert = SessionAlert & { tag: string };

/**
 * アラートContextストアの型定義
 *
 * アラート一覧とアクションを含む
 */
export type AlertsContextStore = Immutable<{
  alerts: TaggedAlert[];
  actions: {
    clearAlert: (tag: string) => void;
    setAlert: (tag: string, alert: Immutable<SessionAlert>) => void;
  };
}>;

/**
 * ## AlertsContext
 *
 * **アラート管理のContext**
 *
 * ### 概要
 * - アプリケーション全体のアラート表示を管理
 * - Zustandストアを使用した状態管理
 * - タグベースのアラート識別
 *
 * ### 使用例
 * ```typescript
 * const actions = useAlertsActions();
 *
 * // アラート設定
 * actions.setAlert("error", {
 *   message: "エラーが発生しました",
 *   severity: "error"
 * });
 *
 * // アラート削除
 * actions.clearAlert("error");
 * ```
 */
export const AlertsContext = createContext<undefined | StoreApi<AlertsContextStore>>(undefined);

AlertsContext.displayName = "AlertsContext";

/**
 * アラートストアから値を取得するカスタムフック
 *
 * @param selector - ストアから値を選択する関数
 * @returns T - 選択された値
 */
export function useAlertsStore<T>(selector: (store: AlertsContextStore) => T): T {
  const context = useGuaranteedContext(AlertsContext);
  return useStore(context, selector);
}

const selectActions = (store: AlertsContextStore) => store.actions;

/**
 * アラートアクションを取得するカスタムフック
 *
 * @returns AlertsContextStore["actions"] - アラート操作アクション
 */
export function useAlertsActions(): AlertsContextStore["actions"] {
  return useAlertsStore(selectActions);
}
