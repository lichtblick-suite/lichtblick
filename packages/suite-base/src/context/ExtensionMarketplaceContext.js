// SPDX-FileCopyrightText: Copyright (C) 2023-2025 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)<lichtblick@bmwgroup.com>
// SPDX-License-Identifier: MPL-2.0
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/
import { createContext, useContext } from "react";
/**
 * ## ExtensionMarketplaceContext
 *
 * **拡張機能マーケットプレース管理のContext**
 *
 * ### 概要
 * - 拡張機能マーケットプレースへのアクセスを提供
 * - リモートマーケットプレースとの通信を管理
 * - 拡張機能の検索・取得・情報表示を統合
 *
 * ### マーケットプレース機能
 * - **拡張機能カタログ**: 利用可能な拡張機能の一覧表示
 * - **検索・フィルタ**: 目的別の拡張機能検索
 * - **詳細情報**: 拡張機能の説明・使用方法・評価
 * - **セキュリティ情報**: ハッシュ・署名・検証状態
 * - **バージョン履歴**: 過去のバージョンと変更履歴
 *
 * ### ユーザー体験
 * - **ブラウジング**: 拡張機能の閲覧・比較
 * - **プレビュー**: インストール前の機能確認
 * - **レビュー**: ユーザー評価・コメント
 * - **推奨機能**: 関連拡張機能の提案
 *
 * @see ExtensionMarketplace - マーケットプレースインターフェース
 * @see ExtensionCatalog - 拡張機能カタログ
 * @see ExtensionInfo - 拡張機能基本情報
 */
const ExtensionMarketplaceContext = createContext(undefined);
ExtensionMarketplaceContext.displayName = "ExtensionMarketplaceContext";
/**
 * ## useExtensionMarketplace
 *
 * **拡張機能マーケットプレースにアクセスするためのカスタムフック**
 *
 * ### 概要
 * - ExtensionMarketplaceContextからExtensionMarketplaceインスタンスを取得
 * - マーケットプレースの検索・取得機能を提供
 * - 必須のContext依存関係をチェック
 *
 * ### 使用例
 * ```typescript
 * function ExtensionStoreComponent() {
 *   const marketplace = useExtensionMarketplace();
 *   const [extensions, setExtensions] = React.useState<ExtensionMarketplaceDetail[]>([]);
 *   const [loading, setLoading] = React.useState(false);
 *
 *   // 拡張機能一覧の取得
 *   React.useEffect(() => {
 *     const fetchExtensions = async () => {
 *       setLoading(true);
 *       try {
 *         const available = await marketplace.getAvailableExtensions();
 *         setExtensions(available);
 *       } catch (error) {
 *         console.error("Failed to fetch extensions:", error);
 *       } finally {
 *         setLoading(false);
 *       }
 *     };
 *
 *     fetchExtensions();
 *   }, [marketplace]);
 *
 *   // 拡張機能の詳細情報表示
 *   const handleShowDetails = async (extension: ExtensionMarketplaceDetail) => {
 *     if (extension.readme) {
 *       try {
 *         const readmeContent = await marketplace.getMarkdown(extension.readme);
 *         // README表示ダイアログを開く
 *         showReadmeDialog(readmeContent);
 *       } catch (error) {
 *         console.error("Failed to fetch README:", error);
 *       }
 *     }
 *   };
 *
 *   // 拡張機能のインストール
 *   const handleInstall = async (extension: ExtensionMarketplaceDetail) => {
 *     if (!extension.foxe) {
 *       console.error("No download URL available");
 *       return;
 *     }
 *
 *     try {
 *       // ExtensionCatalogを使用してインストール
 *       const catalog = useExtensionCatalog((state) => state);
 *       const extensionData = await catalog.downloadExtension(extension.foxe);
 *
 *       // ハッシュ検証（利用可能な場合）
 *       if (extension.sha256sum) {
 *         const isValid = await verifyHash(extensionData, extension.sha256sum);
 *         if (!isValid) {
 *           throw new Error("Hash verification failed");
 *         }
 *       }
 *
 *       const results = await catalog.installExtensions("user", [extensionData]);
 *       if (results[0]?.success) {
 *         console.log("Extension installed successfully");
 *       }
 *     } catch (error) {
 *       console.error("Installation failed:", error);
 *     }
 *   };
 *
 *   if (loading) {
 *     return <div>Loading extensions...</div>;
 *   }
 *
 *   return (
 *     <div>
 *       <h2>Extension Marketplace</h2>
 *       <div className="extension-grid">
 *         {extensions.map(ext => (
 *           <div key={ext.id} className="extension-card">
 *             <h3>{ext.name}</h3>
 *             <p>{ext.description}</p>
 *             <div className="extension-meta">
 *               <span>Version: {ext.version}</span>
 *               <span>Author: {ext.author}</span>
 *               {ext.sha256sum && (
 *                 <span className="verified">✓ Verified</span>
 *               )}
 *             </div>
 *             <div className="extension-actions">
 *               <button onClick={() => handleShowDetails(ext)}>
 *                 Details
 *               </button>
 *               <button onClick={() => handleInstall(ext)}>
 *                 Install
 *               </button>
 *             </div>
 *           </div>
 *         ))}
 *       </div>
 *     </div>
 *   );
 * }
 *
 * // 拡張機能検索コンポーネント
 * function ExtensionSearchComponent() {
 *   const marketplace = useExtensionMarketplace();
 *   const [searchTerm, setSearchTerm] = React.useState("");
 *   const [filteredExtensions, setFilteredExtensions] = React.useState<ExtensionMarketplaceDetail[]>([]);
 *
 *   const handleSearch = async () => {
 *     const allExtensions = await marketplace.getAvailableExtensions();
 *     const filtered = allExtensions.filter(ext =>
 *       ext.name.toLowerCase().includes(searchTerm.toLowerCase()) ||
 *       ext.description?.toLowerCase().includes(searchTerm.toLowerCase()) ||
 *       ext.keywords?.some(keyword =>
 *         keyword.toLowerCase().includes(searchTerm.toLowerCase())
 *       )
 *     );
 *     setFilteredExtensions(filtered);
 *   };
 *
 *   return (
 *     <div>
 *       <input
 *         type="text"
 *         value={searchTerm}
 *         onChange={(e) => setSearchTerm(e.target.value)}
 *         placeholder="Search extensions..."
 *       />
 *       <button onClick={handleSearch}>Search</button>
 *
 *       <div className="search-results">
 *         {filteredExtensions.map(ext => (
 *           <div key={ext.id}>
 *             {ext.name} - {ext.description}
 *           </div>
 *         ))}
 *       </div>
 *     </div>
 *   );
 * }
 * ```
 *
 * ### セキュリティ考慮事項
 * - 拡張機能のダウンロード前にハッシュ検証を実行
 * - 信頼できるマーケットプレースのみからダウンロード
 * - ユーザーに対してセキュリティ状態を明示
 * - 不正な拡張機能の検出と警告
 *
 * ### エラーハンドリング
 * - ネットワークエラーの適切な処理
 * - マーケットプレース接続失敗時のフォールバック
 * - 拡張機能情報の取得失敗時の代替表示
 *
 * @returns {ExtensionMarketplace} 拡張機能マーケットプレースインターフェース
 * @throws {Error} ExtensionMarketplaceProviderが設定されていない場合
 */
export function useExtensionMarketplace() {
    const extensionMarketplace = useContext(ExtensionMarketplaceContext);
    if (extensionMarketplace == undefined) {
        throw new Error("An ExtensionMarketplaceContext provider is required to useExtensionMarketplace");
    }
    return extensionMarketplace;
}
export default ExtensionMarketplaceContext;
