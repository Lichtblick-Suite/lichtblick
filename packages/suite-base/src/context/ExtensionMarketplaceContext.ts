// This Source Code Form is subject to the terms of the Mozilla Public
// License, v2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/

import { ExtensionNamespace } from "@lichtblick/suite-base/types/Extensions";
import { createContext, useContext } from "react";

export type ExtensionMarketplaceDetail = {
  id: string;
  name: string;
  qualifiedName: string;
  namespace?: ExtensionNamespace;
  description: string;
  publisher: string;
  homepage: string;
  license: string;
  version: string;
  readme?: string;
  changelog?: string;
  sha256sum?: string;
  foxe?: string;
  keywords?: string[];
  time?: Record<string, string>;
};

export interface ExtensionMarketplace {
  getAvailableExtensions(): Promise<ExtensionMarketplaceDetail[]>;
  getMarkdown(url: string): Promise<string>;
}

const ExtensionMarketplaceContext = createContext<ExtensionMarketplace | undefined>(undefined);
ExtensionMarketplaceContext.displayName = "ExtensionMarketplaceContext";

export function useExtensionMarketplace(): ExtensionMarketplace {
  const extensionMarketplace = useContext(ExtensionMarketplaceContext);
  if (extensionMarketplace == undefined) {
    throw new Error(
      "An ExtensionMarketplaceContext provider is required to useExtensionMarketplace",
    );
  }
  return extensionMarketplace;
}

export default ExtensionMarketplaceContext;
