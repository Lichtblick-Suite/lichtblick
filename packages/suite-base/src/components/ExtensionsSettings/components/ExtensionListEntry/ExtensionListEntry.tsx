// SPDX-FileCopyrightText: Copyright (C) 2023-2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)<lichtblick@bmwgroup.com>
// SPDX-License-Identifier: MPL-2.0

import { ListItem, ListItemButton, ListItemText, Typography } from "@mui/material";

import { Immutable } from "@lichtblick/suite";
import Stack from "@lichtblick/suite-base/components/Stack";
import TextHighlight from "@lichtblick/suite-base/components/TextHighlight";
import { ExtensionMarketplaceDetail } from "@lichtblick/suite-base/context/ExtensionMarketplaceContext";

import { useStyles } from "./ExtensionListEntry.style";

type Props = {
  entry: Immutable<ExtensionMarketplaceDetail>;
  onClick: () => void;
  searchText: string;
};

export default function ExtensionListEntry({
  entry: { id, description, name, publisher, version },
  searchText,
  onClick,
}: Props): React.JSX.Element {
  const { classes } = useStyles();
  return (
    <ListItem disablePadding key={id}>
      <ListItemButton className={classes.listItemButton} onClick={onClick}>
        <ListItemText
          disableTypography
          primary={
            <Stack direction="row" alignItems="baseline" gap={0.5}>
              <Typography variant="subtitle2" fontWeight={600}>
                <TextHighlight targetStr={name} searchText={searchText} />
              </Typography>
              <Typography variant="caption" color="text.secondary">
                {version}
              </Typography>
            </Stack>
          }
          secondary={
            <Stack gap={0.5}>
              <Typography variant="body2" color="text.secondary">
                {description}
              </Typography>
              <Typography color="text.primary" variant="body2">
                {publisher}
              </Typography>
            </Stack>
          }
        />
      </ListItemButton>
    </ListItem>
  );
}
