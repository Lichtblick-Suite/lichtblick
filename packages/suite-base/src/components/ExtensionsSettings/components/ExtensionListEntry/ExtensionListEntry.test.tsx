/** @jest-environment jsdom */

// SPDX-FileCopyrightText: Copyright (C) 2023-2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)<lichtblick@bmwgroup.com>
// SPDX-License-Identifier: MPL-2.0
import { render, screen, fireEvent } from "@testing-library/react";

import { Immutable } from "@lichtblick/suite";
import { ExtensionMarketplaceDetail } from "@lichtblick/suite-base/context/ExtensionMarketplaceContext";
import BasicBuilder from "@lichtblick/suite-base/testing/builders/BasicBuilder";
import "@testing-library/jest-dom";

import ExtensionListEntry from "./ExtensionListEntry";

describe("ExtensionListEntry Component", () => {
  const mockEntry: Immutable<ExtensionMarketplaceDetail> = {
    id: BasicBuilder.string(),
    name: BasicBuilder.string(),
    qualifiedName: BasicBuilder.string(),
    description: BasicBuilder.string(),
    publisher: BasicBuilder.string(),
    homepage: BasicBuilder.string(),
    license: BasicBuilder.string(),
    version: BasicBuilder.string(),
  };

  const mockOnClick = jest.fn();

  it("renders primary text with name and highlights search text", () => {
    render(
      <ExtensionListEntry entry={mockEntry} searchText={mockEntry.name} onClick={mockOnClick} />,
    );

    const name = screen.getByText(new RegExp(mockEntry.name, "i"));
    expect(name).toBeInTheDocument();

    const highlightedText = screen.getByText(new RegExp(mockEntry.name, "i"));
    expect(highlightedText).toBeInTheDocument();
    expect(highlightedText.tagName).toBe("SPAN");
  });

  it("renders secondary text with description and publisher", () => {
    render(
      <ExtensionListEntry entry={mockEntry} searchText={mockEntry.name} onClick={mockOnClick} />,
    );

    const description = screen.getByText(new RegExp(mockEntry.description, "i"));
    expect(description).toBeInTheDocument();

    const publisher = screen.getByText(new RegExp(mockEntry.publisher, "i"));
    expect(publisher).toBeInTheDocument();
  });

  it("displays version next to name", () => {
    render(
      <ExtensionListEntry entry={mockEntry} searchText={mockEntry.name} onClick={mockOnClick} />,
    );

    // Check for version
    const version = screen.getByText(new RegExp(mockEntry.version, "i"));
    expect(version).toBeInTheDocument();
  });

  it("calls onClick when ListItemButton is clicked", () => {
    render(<ExtensionListEntry entry={mockEntry} searchText="" onClick={mockOnClick} />);

    const button = screen.getByRole("button");
    fireEvent.click(button);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
