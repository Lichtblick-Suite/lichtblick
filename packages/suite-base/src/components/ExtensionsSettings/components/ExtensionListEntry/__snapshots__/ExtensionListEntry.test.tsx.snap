// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtensionListEntry Component matches the snapshot 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
  >
    <div
      class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1br4jff-MuiButtonBase-root-MuiListItemButton-root-listItemButton"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <div
          class="css-1fg1ug7-FoxgloveStack-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-h8rzac-MuiTypography-root"
          >
            wtRZDv
          </h6>
          <span
            class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
          >
            ahQlck
          </span>
        </div>
        <div
          class="css-1ucvlxx-FoxgloveStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            CFYPtL
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
          >
            sWzoRy
          </p>
        </div>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </li>
</DocumentFragment>
`;
